### ISSUE TITLE
Failed to create directory ?/.dubbo!

### ISSUE THREAD

**Issue by YZCSZHIWJ** [apache/dubbo#7744]
2.7.7升级2.7.10后出现
java.lang.IllegalArgumentException: Invalid service store file ?/.dubbo/dubbo-metadata-base-service-172.16.0.14-2181.cache, cause: Failed to create directory ?/.dubbo!
        at org.apache.dubbo.metadata.report.support.AbstractMetadataReport.<init>(AbstractMetadataReport.java:113)
        at org.apache.dubbo.metadata.store.zookeeper.ZookeeperMetadataReport.<init>(ZookeeperMetadataReport.java:63)
        at org.apache.dubbo.metadata.store.zookeeper.ZookeeperMetadataReportFactory.createMetadataReport(ZookeeperMetadataReportFactory.java:37)
        at org.apache.dubbo.metadata.report.support.AbstractMetadataReportFactory.getMetadataReport(AbstractMetadataReportFactory.java:49)
        at org.apache.dubbo.metadata.report.MetadataReportFactory$Adaptive.getMetadataReport(MetadataReportFactory$Adaptive.java)
        at org.apache.dubbo.metadata.report.MetadataReportInstance.init(MetadataReportInstance.java:60)
        at org.apache.dubbo.config.bootstrap.DubboBootstrap.startMetadataCenter(DubboBootstrap.java:651)
        at org.apache.dubbo.config.bootstrap.DubboBootstrap.initialize(DubboBootstrap.java:530)
        at org.apache.dubbo.config.bootstrap.DubboBootstrap.start(DubboBootstrap.java:884)
        at org.apache.dubbo.config.spring.context.DubboBootstrapApplicationListener.onContextRefreshedEvent(DubboBootstrapApplicationListener.java:69)
        at org.apache.dubbo.config.spring.context.DubboBootstrapApplicationListener.onApplicationContextEvent(DubboBootstrapApplicationListener.java:62)
        at com.alibaba.spring.context.OnceApplicationContextEventListener.onApplicationEvent(OnceApplicationContextEventListener.java:52)
        at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172)
        at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165)
        at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
        at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:404)
        at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:361)
        at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:898)
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.finishRefresh(ServletWebServerApplicationContext.java:162)
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:554)
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:141)
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:747)
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1215)

-----------------------------------------------------------------------------------------------------------------------------------

环境如下

系统版本：Linux VM_0_14_centos 3.10.0-1062.18.1.el7.x86_64 

docker version
Client: Docker Engine - Community
 Version:           19.03.11
 API version:       1.40
 Go version:        go1.13.10
 Git commit:        42e35e61f3
 Built:             Mon Jun  1 09:13:48 2020
 OS/Arch:           linux/amd64
 Experimental:      false

springboot: 2.2.9.RELEASE

应用使用的docker镜像：openjdk:11

-------------------------------------------------------------------------------------------------------------------------------------

root用户可以启动容器，非root用户报错

2.7.10版本会在容器home下创建.dubbo目录及***.cache文件存放metadata信息，若以非root用户启动容器则会出现上面没有权限的异常

.dubbo目录可以自行配置么，像如下registry.file一样指定，好将其挂载到对应的宿主目录
dubbo:
  registry:
    file: ./data/dubbo-cache

**Comment by BurningCN** [comment]
Brother, you shut it down very quickly, but I took a look

I found this part in the code. Here, 
https://github.com/apache/dubbo/blob/e845e1648f27abbc80828d4b87a7f126eb8d660a/dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/report/support/AbstractMetadataReport.java#L107-L108


the value of the parameter file is taken from the url. This is the same as the one used by the registry. Here
https://github.com/apache/dubbo/blob/e845e1648f27abbc80828d4b87a7f126eb8d660a/dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/AbstractRegistry.java#L101-L102

In other words. If the user does not manually configure the relevant information of the `MetadataReport`, then the information of the `Registry `will be used as the information of the `MetadataReport `by default (see method `useRegistryAsMetadataCenterIfNecessary&registryAsMetadataCenter`). If the user does not configure the `file `in the `registry`, then the `registry `and `MetadataReport `use their default file names. ; If the user adds the `file `parameter in the `registry `(such as` <dubbo:registryaddress="zookeeper://127.0.0.1:2181" file="/home/zs/registry.file"/>`
), then the cache information recorded by the `registry `and `MetadataReport `are all in this file.

The current code does not seem to support specifying `file ` parameter for `MetadataReport`, I will try to add this feature.
